caluma/caluma_core/health_checks.py

```diff
@@ -0,0 +1,213 @@
+from django.conf import settings
+from django.db import connections
+from django.db.migrations.executor import MigrationExecutor
+from django.db.migrations.autodetector import MigrationAutodetector
+from django.db.migrations.state import ProjectState
+from django.apps.registry import apps
+from django.http import JsonResponse
+from django.core import management
+
+from caluma.caluma_logging.models import AccessLog
+from caluma.caluma_form import storage_clients
+
+import watchman
+from watchman import views
+from watchman.decorators import check
+from watchman import settings as watchman_settings
+
+from io import BytesIO, StringIO
+import json
+
+
+@check
+def _check_pending_migrations(db):
+    """
+    Check database for pending migrations.
+    Returns JSON mapping if no pending migrations, otherwise raises Exception.
+    @check decorator catches and handles exception.
+    """
+
+    # check for changes i.e. migrations that haven't been made
+    # makemigrations --check exists with exit status 1 if changes exist
+    try:
+        management.call_command("makemigrations", "--check", "--dry-run")
+    except SystemExit:
+        raise Exception("Migrations aren't up-to-date (makemigrations).")
+
+    # check for unapplied migrations
+    try:
+        out = StringIO()
+        management.call_command("migrate", "--plan", stdout=out)
+        plan = out.getvalue()
+        print(plan)
+        if not "No planned migration operations." in plan:
+            raise Exception("Database has unapplied migrations (migrate).")
+
+        return {db: {"ok": True}}
+
+    finally:
+        # Cleanup
+        out.close()
+
+
+@check
+def _check_minio():
+    """
+    Check minio connection and operations.
+    """
+
+    minio_client = storage_clients.client
+    response = None
+
+    # Check minio operations
+    try:
+        object_name = "my-object"
+        new_object_name = "my-object-new"
```

Comment:
Try to generate request-specific (randomized) object names. If two checks happen at the same time, you may get errors despite the system working correctly. I suggest using the `uuid` module for that. Then, also prefix the object's name so people will know what it is for, if it's not deleted for some reason

Reply:
Added `uuid` to health check. The associated Pytest ran into issues because it couldn't predict the name of the mocked object it was supposed to return. At the moment the health check ignores a wrong object name if it's running in a Pytest. How does one usually solve this kind of problem?

Reply:
Aah, didn't think of that :D I'll send you a link to an internal project that contains code to mock UUID for exactly that purpose.